### PR TITLE
Closes CPPL-198

### PR DESCRIPTION
**Pull request by yaroslav-kernitskyi**
Implemented product view

### CODE REVIEW

File: src/Client/Controller/MainPage/MainPageController.h [hunk 1]
```diff
@@ -33,7 +33,11 @@ class MainPageController : public QObject
 	void InitProductPagesController();
 
 	bool UpdateSubPage(MainSubPages page, PageData data);
+
 	bool UpdateProductsSubPage(PageData data);
```
Comment (Denys-Bykov-0) [inline src/Client/Controller/MainPage/MainPageController.h:37]: Move this to the ProductPagesController

File: src/Client/View/MainPage/Product/ProductViewSubPage/ProductViewSubPage.cpp [hunk 2]
```diff
@@ -30,3 +30,14 @@ ProductViewSubPage::~ProductViewSubPage()
 {
 	delete m_ui;
 }
+
+void ProductViewSubPage::Update(const Product& product)
```
[the remark this answers is not in the document]
Comment (yaroslav-kernitskyi) [reply src/Client/View/MainPage/Product/ProductViewSubPage/ProductViewSubPage.cpp]: Fixed

File: src/Client/Controller/MainPage/MainPageController.h (hunk 1 above)
Comment (yaroslav-kernitskyi) [reply src/Client/Controller/MainPage/MainPageController.h]: Done

**Review by Denys-Bykov-0 — APPROVED** [review]

File: src/Client/Controller/MainPage/MainPageController.cpp [hunk 3]
```diff
@@ -139,14 +148,6 @@ bool MainPageController::UpdateSubPage(MainSubPages page, PageData data)
 	return update_succeeded;
```
Comment (AndriyFedorenko) [inline src/Client/Controller/MainPage/MainPageController.cpp:148]: I don't see reason to have this variable.

File: src/Client/Controller/MainPage/MainPageController.cpp [hunk 4]
```diff
@@ -129,7 +130,15 @@ bool MainPageController::UpdateSubPage(MainSubPages page, PageData data)
 	case MainSubPages::CREATE_LIST:
 		break;
 	case MainSubPages::PRODUCTS:
-		return UpdateProductsSubPage(data);
+		return m_product_pages_controller->UpdateProductsPage(data);
+	case MainSubPages::VIEW_PRODUCT:
+		return m_product_pages_controller->UpdateViewProductSubPage(data);
+	case MainSubPages::CREATE_PRODUCT:
+		return false;
+//		return UpdateCreateProductSubPage(data);
```
Comment (AndriyFedorenko) [inline src/Client/Controller/MainPage/MainPageController.cpp:138]: remove commented code

File: src/Client/Controller/MainPage/ProductPagesController.cpp [hunk 5]
```diff
@@ -8,24 +8,69 @@ ProductPagesController::ProductPagesController(
...
-bool ProductPagesController::UpdateListPage()
+void ProductPagesController::ConnectViewPage()
+{
+	connect(
+		&m_view_page,
+		&ProductViewSubPage::GoBack,
+		this,
+		&ProductPagesController::GoBack);
+
+//	void EditProduct();
```
Comment (AndriyFedorenko) [inline src/Client/Controller/MainPage/ProductPagesController.cpp:59]: the same

File: src/Client/Controller/MainPage/ProductPagesController.cpp [hunk 6]
```diff
@@ -8,24 +8,69 @@ ProductPagesController::ProductPagesController(
...
+		return UpdateProductsPage();
+	}
+	return UpdateProductsPage(qvariant_cast<List>(data));
+}
+
+bool ProductPagesController::UpdateViewProductSubPage(PageData data)
+{
+	if (!data.canConvert<Product>()) {
+		return false;
+		//return m_product_pages_controller->UpdateListPage();
```
Comment (AndriyFedorenko) [inline src/Client/Controller/MainPage/ProductPagesController.cpp:35]: the same

**Review by VasylDobrovolskyi — APPROVED** [review]

**Pull request merged**